Examples/Io/Root/src/RootNTupleReader.cpp

```diff
@@ -0,0 +1,417 @@
+  m_inputChain->SetBranchAddress("truthvertex_y", &m_branches.m_truthvertex_y);
+  m_inputChain->SetBranchAddress("truthvertex_z", &m_branches.m_truthvertex_z);
+  m_inputChain->SetBranchAddress("truthvertex_t", &m_branches.m_truthvertex_t);
+
+  m_inputChain->SetBranchAddress("recovertex_x", &m_branches.m_recovertex_x);
+  m_inputChain->SetBranchAddress("recovertex_y", &m_branches.m_recovertex_y);
+  m_inputChain->SetBranchAddress("recovertex_z", &m_branches.m_recovertex_z);
+  m_inputChain->SetBranchAddress("truthvertex_tracks_idx",
+                                 &m_branches.m_truthvertex_tracks_idx);
+
+  m_inputChain->SetBranchAddress("beamspot_x", &m_branches.m_beamspot_x);
+  m_inputChain->SetBranchAddress("beamspot_y", &m_branches.m_beamspot_y);
+  m_inputChain->SetBranchAddress("beamspot_z", &m_branches.m_beamspot_z);
+  m_inputChain->SetBranchAddress("beamspot_sigX", &m_branches.m_beamspot_sigX);
+  m_inputChain->SetBranchAddress("beamspot_sigY", &m_branches.m_beamspot_sigY);
+  m_inputChain->SetBranchAddress("beamspot_sigZ", &m_branches.m_beamspot_sigZ);
+
+  auto path = m_cfg.inputFilePath;
+
+  // add file to the input chain
+  m_inputChain->Add(path.c_str());
+  ACTS_DEBUG("Adding File " << path << " to tree '" << m_cfg.inputTreeName
+                            << "'.");
+
+  m_events = m_inputChain->GetEntries();
+  ACTS_DEBUG("The full chain has " << m_events << " entries.");
+
+  // If the events are not in order, get the entry numbers for ordered events
+  m_entryNumbers.resize(m_events);
+  m_inputChain->Draw("EventNumber", "", "goff");
+  // Sort to get the entry numbers of the ordered events
+  TMath::Sort(m_inputChain->GetEntries(), m_inputChain->GetV1(),
+              m_entryNumbers.data(), false);
+}
+
+std::pair<std::size_t, std::size_t>
+ActsExamples::RootNTupleReader::availableEvents() const {
+  return {0u, m_events};
+}
+
+ActsExamples::ProcessCode ActsExamples::RootNTupleReader::read(
+    const ActsExamples::AlgorithmContext& context) {
+  ACTS_DEBUG("Trying to read track parameters from ntuple.");
+
+  Acts::Vector3 pos(0, 0, 0);
+  std::shared_ptr<Acts::PerigeeSurface> surface =
+      Acts::Surface::makeShared<Acts::PerigeeSurface>(pos);
+
+  if (context.eventNumber >= m_events) {
+    ACTS_ERROR("event out of bounds");
+  }
+
+  std::lock_guard<std::mutex> lock(m_read_mutex);
+
+  auto entry = context.eventNumber;
+  m_inputChain->GetEntry(entry);
+  ACTS_INFO("Reading event: " << context.eventNumber
+                              << " stored as entry: " << entry);
+
+  const unsigned int nTracks = m_branches.m_track_d0->size();
+  const unsigned int nTruthVtx = m_branches.m_truthvertex_z->size();
+  const unsigned int nRecoVtx = m_branches.m_recovertex_z->size();
+
+  ACTS_DEBUG("nTracks = " << nTracks);
+  ACTS_DEBUG("nTruthVtx = " << nTruthVtx);
+  ACTS_DEBUG("nRecoVtx = " << nRecoVtx);
+
+  std::vector<Acts::BoundTrackParameters> trackContainer;
```

Comment:
I wonder if we should use the typedefs like `ActsExamples::TrackParametersContainer` etc. to stick to the convention.
Same below for the other vectors if possible?

Reply:
makes sense :+1: 

the other vector would be `std::vector<Acts::Vector4>`. do we have a common typedef for these?

Reply:
I don't know of one, I would just leave it then.